### PR TITLE
logik_demo architecture documentation

### PR DESCRIPTION
**Pull request by petergrossmann21**
Startpoint for documenting the architecture.  Intended to be a light overview, so figuring out whether it needs to grow/shrink to meet that objective is of particular interest.

### CODE REVIEW

File: docs/source/logik_demo.rst [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+--------------------------
+
+The following sections briefly describe the logic resources of the logik_demo eFPGA at a high level.  Emphasis is placed on topics that are useful in studying the outputs of flow steps in Logik.
+
+Lookup Tables and Registers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+logik_demo contains 4-input lookup tables (LUTs) for implementing general purpose logic.  During logic synthesis, digital logic is mapped to these LUTs.
+
+Each LUT is paired with a D flip-flop to form a basic logic element (BLE).  BLE inputs and LUT inputs are wired together and thus equivalent.  Each flip flop can be configured with or without any of the following features:  asynchronous set, asynchronous reset, built-in enable bit.  Logic synthesis determines how many flip-flops are needed of each configuration type.  Placement and routing determine whether or not to pair LUT with its flip-flop.  Ifn the lut is paired, the BLE output receives the flip-flop output; otherwise, it receives the LUT output.  The BLE output interfaces to the FPGA's programmable interconnect.
```
Comment (sgherbst) [inline docs/source/logik_demo.rst:32]: typo: ifn

File: docs/source/logik_demo.rst [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+Multiply-Add Engine (MAE)
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The multiply-add engine (MAE) is a configurable arithmetic unit suitable for use in many digital signal processing (DSP) applications.  The key blocks are a multiplier and an adder, which can be used one at a time or together as a multiply-accumulate (MAC) unit.  An 18x18 multiplier receives data directly from the ebrick-fpga global interconnect or optionally from flip-flops to re-register data for improved throughput.  The output of the multiplier can be routed directly out of the MAE to global interconnect to realize a purely combinational multiplier circuit or routed through a bank of flip-flops.  In either case, this output can instead be routed through a 40-bit accumulator so that the MAE can be used as a MAC rather than a multiplier.  Both the multiplier and accumulator can be registered to create a pipelined MAC.  The accumulator can also be used by itself, in which case MAE inputs are routed directly to it.
```
Comment (sgherbst) [inline docs/source/logik_demo.rst:39]: logik_demo instead of ebrick-fpga?

File: docs/source/logik_demo.rst [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
++-------------+----------------------+----------------------------------+----------------+
+| Accumulator | efpga_macc_pipe      | Inputs unregistered              | 1              | 
++-------------+----------------------+----------------------------------+----------------+
+| Accumulator | efpga_macc_pipe_regi | Inputs registered                | 2              |
++-------------+----------------------+----------------------------------+----------------+
+
+Block RAMs (BRAMs)
+^^^^^^^^^^^^^^^^^^
+
+Each block RAM (BRAM) consists 8KB of single-port SRAM, organized as 2K 32-bit words.  The 8KB of SRAM can be used to emulate 4KB of simple dual-port SRAM with one write port, one read port, and one common clock.  It may also be used to emulate 1K 64-bit words when configured as single-port RAM.
```
Comment (sgherbst) [inline docs/source/logik_demo.rst:79]: typo: consists 8KB

**Review by sgherbst — APPROVED** [review]
Looks great!  Both clear and thorough.  Only suggestion would be to add diagrams for the BLE, CLB, etc.

**Review by volkamir — APPROVED** [review]
Looks great!

**Pull request merged**